router/pcap.go

```diff
@@ -1,19 +1,39 @@
 package router
 
 import (
-	"fmt"
-
 	"code.google.com/p/gopacket/pcap"
+	"fmt"
+	"net"
+	"sync"
 )
 
-type PcapIO struct {
-	handle *pcap.Handle
+type Pcap struct {
+	iface *net.Interface
+	bufSz int
+
+	// The libpcap handle for writing packets. It's possible thata
+	// single handle could be used for reading and writing, but
+	// we'd have to examine the performance implications.
+	writeHandle *pcap.Handle
+
+	// pcap handles are single-threaded, so we need to lock around
+	// uses of writeHandle.
+	writeHandleMutex sync.Mutex
```

Comment:
I've been staring at the code for half an hour now and still cannot work out where the multiple goroutines come from that call `Send`. Or did you introduce the lock to make the API thread-safe *just in case*?